main/states/idle_state.py

```diff
@@ -43,11 +44,33 @@ def on_enter(self, payload=None):
         self.notify_renderer('idle')
 
     def __detected(self):
-        if self.isActive:
+        if hasattr(self, 'video_process') and self.video_process != None:
+            self.video_process.send_signal(signal.SIGSTOP)  # nosec #pylint-disable type: ignore
+            lights.off()
+            lights.wakeup()
             subprocess.Popen(['play', os.path.join(self.components.config['data_base_dir'],
-                                                   self.components.config['detection_bell_sound'])])  # nosec # pylint-disable type: ignore
-            self.transition(state=self.allowedStateTransitions.get(
-                'recognizing'), payload=None)
+                                                   self.components.config['detection_bell_sound'])])  # nosec #pylint-disable type: ignore
+            lights.wakeup()
+            self.transition(self.allowedStateTransitions.get('recognizing'))
+            self.video_process.send_signal(signal.SIGCONT)  # nosec #pylint-disable type: ignore
+
+        elif hasattr(self, 'audio_process') and self.audio_process != None:
+            self.audio_process.send_signal(signal.SIGSTOP)  # nosec #pylint-disable type: ignore
+            lights.off()
+            lights.wakeup()
+            subprocess.Popen(['play', os.path.join(self.components.config['data_base_dir'],
+                                                   self.components.config['detection_bell_sound'])])  # nosec #pylint-disable type: ignore
+            lights.wakeup()
+            self.transition(self.allowedStateTransitions.get('recognizing'))
+            self.audio_process.send_signal(signal.SIGCONT)  # nosec #pylint-disable type: ignore
+
```

Comment:
Can we get rid of this duplication of code? Besides the process that is the signal sent to it is the same code, so it shouldn't be duplicated.

Reply:
updated